crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -387,27 +387,30 @@ where
     ///
     /// If this is the first time the namespace is requested, a new instance of API implementation
     /// will be created.
-    pub fn reth_methods(&mut self, namespace: RethRpcModule) -> Methods {
-        if let Some(methods) = self.modules.get(&namespace).cloned() {
-            return methods
-        }
-        let methods: Methods = match namespace {
-            RethRpcModule::Admin => AdminApi::new(self.network.clone()).into_rpc().into(),
-            RethRpcModule::Debug => {
-                let eth_api = self.eth_api();
-                DebugApi::new(eth_api).into_rpc().into()
-            }
-            RethRpcModule::Eth => self.eth_api().into_rpc().into(),
-            RethRpcModule::Net => {
-                let eth_api = self.eth_api();
-                NetApi::new(self.network.clone(), eth_api).into_rpc().into()
-            }
-            RethRpcModule::Trace => TraceApi::new().into_rpc().into(),
-            RethRpcModule::Web3 => Web3Api::new(self.network.clone()).into_rpc().into(),
-        };
-        self.modules.insert(namespace, methods.clone());
-
-        methods
+    pub fn reth_methods(
+        &mut self,
+        namespaces: impl Iterator<Item = RethRpcModule>,
+    ) -> Vec<Methods> {
+        let eth_api = self.eth_api();
```

Comment:
this will always create an `EthApi` instance even though it might not be used.

It will not register it, if not required. 
This is fine and perhaps even comes in handy when we need to returns the EthAPI somehow